### DISCUSSION TITLE
Windscribe Static IP Support

### DISCUSSION THREAD

**Discussion by avLightless** [qdm12/gluetun#592] [Ideas]
Currently it is not possible to choose your static ip for windscribe as it is not any of the lists. Adding a custom ip via the openvpn settings also does not work. Port forwarding at windscribe only works with static ips, so adding this feature would enable port forwarding for users.

**Comment by qdm12** [comment]
Can you use `OPENVPN_TARGET_IP` for that?

EDIT: If not, where do you get those IP addresses from, I can add them to the built-in servers list.

**Reply by avLightless** [reply]
No, when I use `OPENVPN_TARGET_IP` I get this error message `gluetun       | 2021/08/27 15:23:21 ERROR vpn: failed building configuration: target IP address not found: in 686 filtered connections`.
I tried adding an entry manually to servers.json but it seems to get overwritten.
I don't think the list of static ips is public, windscribe just gives you one based on the location you select when you subscribe.

**Reply by qdm12** [reply]
I'll change the program behavior to just use the IP address given without checking if it exists.

Regarding changing the servers.jjson manually, it's another bug, you can subscribe to #566 to know when it solved.

**Reply by qdm12** [reply]
119cac5a670c70aa3c48c8533e11a280e482eaba changes `OPENVPN_TARGET_IP` to now override the IP address from the connection picked, and uses the check for existing servers only when using Wireguard (since we need the corresponding public key of the server). That should give you a workaround for now until #566 is fixed so you can add servers manually.
